txtorcon/torinfo.py

```diff
@@ -99,6 +109,9 @@ def stripper(key, arg):
 
         d = self.proto.get_info_raw(req)
         d.addCallback(functools.partial(stripper, req))
+        descriptor_keys = ["desc/name", "desc/id"]
+        if self.info_key in descriptor_keys and self.proto.use_stem:
+            d.addCallback(_parse_stem_descriptor)
         return d
```

Comment:
I meant that we can probably just use the "RelayDescriptor" constructor as the callback, directly (since all we do in _parse_stem_descriptor is call RelayDesciptor(data)). Might actually be better to just leave the method you have, though, in case it needs more stuff in future ...?